### ISSUE TITLE
Segmentation Fault when running an app that uses grpc, numpy and sqlite.

### ISSUE THREAD

**Issue by alessandrocarrese** [Nuitka/Nuitka#3271]
- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.

  > python -m nuitka --version

```
2.5.7
Commercial: None
Python: 3.13.0 (v3.13.0:60403a5409f, Oct  7 2024, 00:37:40) [Clang 15.0.0 (clang-1500.3.9.4)]
Flavor: CPython Official
GIL: yes
Executable: /Library/Frameworks/Python.framework/Versions/3.13/bin/python3
OS: Darwin
Arch: arm64
macOSRelease: 14.4
Version C compiler: /usr/bin/clang (clang 15.0.0).
```

- How did you install Nuitka and Python

`  I installed it using pip and I tried both with virtual env and not.
`
- The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.

  > python -m pip list -v

```
altgraph                  0.17.4      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
Cython                    3.0.11      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
distlib                   0.3.9       /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
filelock                  3.16.1      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
grpcio                    1.68.1      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
macholib                  1.16.3      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
Nuitka                    2.5.7       /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
numpy                     2.1.3       /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
ordered-set               4.1.0       /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
packaging                 24.2        /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
pandas                    2.2.3       /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
pip                       24.2        /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
platformdirs              4.3.6       /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
protobuf                  5.29.0      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
pyinstaller               6.11.1      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
pyinstaller-hooks-contrib 2024.10     /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
python-dateutil           2.9.0.post0 /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
pytz                      2024.2      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
setuptools                75.6.0      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
six                       1.16.0      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
tzdata                    2024.2      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
virtualenv                20.28.0     /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
zstandard                 0.23.0      /Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages pip
```

- Many times when you get an error from Nuitka, your setup may be special

The app is working correctly when started using regular python.

- Also supply a Short, Self Contained, Correct, Example

```
  The app starts a grpc server and connect to a database (sqlite).
  It is expecting some parameters to be passed via sys args when the executables is started.
```

- Avoid unnecessary options

`Tried with --standalone and --onefile, both and alone
`

**Comment by kayhayen** [comment]
Without a MRE this is interesting, but useless on its own, probably one of many 3.13 problems to find triggers it, but we need them in isolation.
